### PR TITLE
Community/add contribution guidelines

### PR DESCRIPTION
**Pull request by svettwer**
Hi!

This is my suggestion for a first version of contribution guidelines.
I'm looking forward to your feedback!

BR,
Sven

P.S. Because this is just a `.md` file, I skipped the travis build :wink:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+This document contains a set of guidelines for contributing to the Citrus framework ecosystem.
+The intention of these guidelines is to make it easier to work together. Nevertheless these guidelines are just
+suggestions and not set in stone. Use them to the best of your knowledge and feel free to propose changes in a
+pull request. 
+
+_Credit: This Contribution guidelines are based on the
+[Atom contribution guidelines](https://github.com/atom/atom/blob/master/CONTRIBUTING.md)_
+
+> Note: If you want your contributions in the official Citrus code repository you have to agree that your changes are
```
Comment (martinmaher) [inline CONTRIBUTING.md]: > Note: In order to have your contributions included in the official Citrus code repository you have to agree to release your changes under the same licensing agreement that Citrus uses at that time. Please also note that the Citrus team might change the licensing of Citrus some time in future.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  * [Suggesting Changes](#suggesting-changes)
+  * [Contributions](#contributions)
+  * [Pull Requests](#pull-requests)
+  * [Types of change](#types-of-change)
+  * [Definition of ready](#definition-of-ready)
+  * [Definition of done](#definition-of-done)
+
+## I just have a question
+
+Before you open an issue concerning your question, please make sure that none of the following steps led to an answer.
```
Comment (martinmaher) [inline CONTRIBUTING.md]: .. following steps already lead to an answer.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+* [Search for issues](https://github.com/citrusframework/citrus/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22Type%3A+Bug%22+)
+  to see if the problem has already been reported. If it has **and the issue is still open**, add a comment to the
+  existing issue instead of opening a new one.
+
+
+### Suggesting changes
+
+This section guides you through submitting enhancement suggestions, feature requests or maintenance tasks for Citrus.
+Following these guidelines helps maintainers and the community understand your suggestion :pencil: and find related
+suggestions :mag_right:. We distinguish between different [types of change](#types-of-change):
```
Comment (martinmaher) [inline CONTRIBUTING.md]: .. different types of changes:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+[the template](https://github.com/citrusframework/citrus/issues/new?template=feature_request.md) and ensure, that the 
+[definition of ready](#definition-of-ready) is fulfilled.
+
+### Contributions
+Please feel free to begin with any issue you would like to work on. It might be helpful to get some orientation by having
+a look at our [types of change](#types-of-change) and the estimated requirements linked to them.  
+
+>**Important Note**: If you'd like to contribute a bug fix, please ensure to branch from `v2.7-bugfix` instead of master.
+
+We also always high appreciate contributions to milestones and we love to work closely with the community.
```
Comment (martinmaher) [inline CONTRIBUTING.md]: We also highly appreciate contributions to milestones ..

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+* The code should have a suitable test coverage
+* The changes should be documented
+
+### Types of change
+We distinguish between different types of change which indicate different level of complexity, scope and required
+knowledge about the Citrus framework.
+ 
+* **Maintenance**  
+  Tasks required to keep the framework up to date. E.g. updating dependencies. Depending on the tasks, 
+  the changes may vary from smaller to larger size with a according time required. Most of the tasks just require a
```
Comment (martinmaher) [inline CONTRIBUTING.md]: (Depending on the tasks, the changes may vary from smaller to larger size with a according time required)
Tasks may vary in size from smaller to larger changes.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+The definition of ready (DOR) specifies the entry-criteria for an issue that is currently in in the backlog to be moved
+to the *ready* state so that it's possible to work on it. If you open a new issue, it would be great if you could ensure
+that the following criteria are met. This helps us to understand your issue and allows to start developing right away
+without any further research/reconstruction efforts.
+
+* The corresponding issue template for [bugs](https://github.com/citrusframework/citrus/issues/new?template=bug_report.md)
+  or [features](https://github.com/citrusframework/citrus/issues/new?template=feature_request.md) has been filled out completely
+* Bigger changes have been transformed to epics and were broken down into smaller issues
+* The context of the issue is understood by at least one maintainer
+  * Therefore it's important to provide as many context as possible
```
Comment (martinmaher) [inline CONTRIBUTING.md]: Therefore it's important to provide as much context as possible

**Comment by svettwer** [comment]
Hey Martin!

Awesome! :+1: 
Thx a lot for your feedback!

**Comment by svettwer** [comment]
I'll just merge as I Just added the outstanding review changes.

**Pull request merged**